core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/UniqueIDUserStoreManager.java

```diff
@@ -543,17 +508,54 @@ void updateGroupListOfUser(String userID, String[] deletedGroupIDs, String[] new
      */
     Group renameGroup(String groupID, String newGroupName) throws UserStoreException;
 
+    // User centric group related implementation methods ..............................................................
+
     /**
      * Add a user.
      *
      * @param userName    User Name.
      * @param credential  Credentials.
      * @param claims      Maps of user claim values.
-     * @param groupIDs    Array of group IDs.
+     * @param groupIDs    List of group IDs.
      * @param profileName Profile name.
      * @return User object.
      * @throws UserStoreException thrown by the underlying UserStoreManager.
      */
-    User addUser(String userName, Object credential, Map<String, String> claims, String[] groupIDs,
+    User addUser(String userName, Object credential, Map<String, String> claims, List<String> groupIDs,
                  String profileName) throws UserStoreException;
+
+    /**
+     * Retrieves list of groups of a given user ID.
+     *
+     * @param userId    User ID.
+     * @param limit     No of search results. If the given value is greater than the system configured max limit
```

Comment:
```suggestion
     * @param limit     No of search results. If the given value is greater than the system configured max limit.
```

Reply:
this comment has more in the next line, hence not put fullstop